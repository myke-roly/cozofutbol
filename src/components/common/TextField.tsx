import React from 'react'
import {
  KeyboardTypeOptions,
  StyleSheet,
  Text,
  TextInput,
  TextInputProps,
  View,
} from 'react-native'

type TextFieldProps = {
  value: string
  onChange: (value: string) => void
  placeholder: string
  label: string
  type?: KeyboardTypeOptions
} & TextInputProps

const TextField = ({
  placeholder,
  onChange,
  value,
  label,
  type = 'default',
  ...props
}: TextFieldProps) => {
  return (
    <View>
      <Text>{label}</Text>
      <TextInput
        style={styles.input}
        {...props}
        placeholder={placeholder}
        keyboardType={type}
        value={value}
        onChangeText={onChange}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  input: {
    borderWidth: 1,
    borderColor: 'gray',
    padding: 8,
    borderRadius: 8,
  },
})

export default TextField
