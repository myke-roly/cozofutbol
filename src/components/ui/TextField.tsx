import React, { forwardRef, memo } from 'react'
import { KeyboardTypeOptions, StyleSheet, TextInput } from 'react-native'
import { Color, FontSize } from '../../constants'

type TextFieldProps = {
  value: string
  onChange: (value: string) => void
  placeholder: string
  type?: KeyboardTypeOptions
  divider?: boolean
}

const TextField = forwardRef<TextInput, TextFieldProps>(
  (
    {
      placeholder,
      divider,
      onChange,
      value = '',
      type = 'default',
      ...props
    }: TextFieldProps,
    ref,
  ) => {
    console.log(
      'render TextField',
      //typeof ref === 'function' ? ref(null) : ref?.current,
    )
    return (
      <TextInput
        ref={ref}
        style={[styles.input, divider && styles.divider]}
        placeholder={placeholder}
        placeholderTextColor={Color.GRAY}
        keyboardType={type}
        value={value}
        onChangeText={onChange}
        autoCapitalize="none"
        {...props}
      />
    )
  },
)

const styles = StyleSheet.create({
  input: {
    borderWidth: 0,
    paddingHorizontal: 16,
    paddingVertical: 12,
    fontSize: FontSize.MEDIUM,
    fontWeight: '500',
    color: Color.BLACK,
  },
  divider: {
    borderBottomWidth: 1,
    borderBottomColor: Color.GRAY,
  },
})

export default memo(TextField)
