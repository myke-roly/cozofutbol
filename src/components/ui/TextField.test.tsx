import renderer from 'react-test-renderer'
import TextField from './TextField'

describe('TextField', () => {
  const fn = jest.fn()
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <TextField
          value="value-test"
          onChange={fn}
          placeholder="placeholder-test"
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with optionals props', () => {
    const tree = renderer
      .create(
        <TextField
          value="value-test"
          onChange={fn}
          placeholder="placeholder-test"
          divider={true}
          type="number-pad"
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
