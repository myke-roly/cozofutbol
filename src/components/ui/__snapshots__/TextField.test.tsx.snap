// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should render correctly 1`] = `
<TextInput
  autoCapitalize="none"
  keyboardType="default"
  onChangeText={[MockFunction]}
  placeholder="placeholder-test"
  placeholderTextColor="#acacac"
  style={
    [
      {
        "borderWidth": 0,
        "color": "#232323",
        "fontSize": 16,
        "fontWeight": "500",
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      },
      undefined,
    ]
  }
  value="value-test"
/>
`;

exports[`TextField should render correctly with optionals props 1`] = `
<TextInput
  autoCapitalize="none"
  keyboardType="number-pad"
  onChangeText={[MockFunction]}
  placeholder="placeholder-test"
  placeholderTextColor="#acacac"
  style={
    [
      {
        "borderWidth": 0,
        "color": "#232323",
        "fontSize": 16,
        "fontWeight": "500",
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      },
      {
        "borderBottomColor": "#acacac",
        "borderBottomWidth": 1,
      },
    ]
  }
  value="value-test"
/>
`;
